[torchgen] Let native function declaration generation logic take a callable

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89595
* __->__ #89594
* #89487

This PR allows `get_native_function_declarations` API to take a function as argument. This function should take `NativeFunction` as input and emit code for native function declaration. By default it is `dest.compute_native_function_declaration`.

Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

larryliu0820 committed:
[torchgen] Let native function declaration generation logic take a callable



Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[torchgen] Let native function declaration generation logic take a callable"


This PR allows `get_native_function_declarations` API to take a function as argument. This function should take `NativeFunction` as input and emit code for native function declaration. By default it is `dest.compute_native_function_declaration`.

Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[torchgen] Let native function declaration generation logic take a callable"


This PR allows `get_native_function_declarations` API to take a function as argument. This function should take `NativeFunction` as input and emit code for native function declaration. By default it is `dest.compute_native_function_declaration`.

Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41501838).

larryliu0820 committed:
Update on "[torchgen] Let native function declaration generation logic take a callable"


This PR allows `get_native_function_declarations` API to take a function as argument. This function should take `NativeFunction` as input and emit code for native function declaration. By default it is `dest.compute_native_function_declaration`.

Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[torchgen] Let native function declaration generation logic take a callable"


This PR allows `get_native_function_declarations` API to take a function as argument. This function should take `NativeFunction` as input and emit code for native function declaration. By default it is `dest.compute_native_function_declaration`.

Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[torchgen] Let native function declaration generation logic take a callable"


This PR allows `get_native_function_declarations` API to take a function as argument. This function should take `NativeFunction` as input and emit code for native function declaration. By default it is `dest.compute_native_function_declaration`.

Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[torchgen] Let native function declaration generation logic take a callable"


This PR allows `get_native_function_declarations` API to take a function as argument. This function should take `NativeFunction` as input and emit code for native function declaration. By default it is `dest.compute_native_function_declaration`.

Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

[ghstack-poisoned]

ezyang commented on torchgen/gen.py:
```diff
@@ -1406,6 +1417,9 @@ def get_native_function_declarations(
     backend_indices: Dict[DispatchKey, BackendIndex],
+    native_function_decl_gen: Callable[
+        [Union[NativeFunctionsGroup, NativeFunction], BackendIndex], List[str]
+    ],
```
A comment about what this is for and how to use it would be good

larryliu0820 committed:
Update on "[torchgen] Let native function declaration generation logic take a callable"


This PR allows `get_native_function_declarations` API to take a function as argument. This function should take `NativeFunction` as input and emit code for native function declaration. By default it is `dest.compute_native_function_declaration`.

Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[torchgen] Let native function declaration generation logic take a callable"


This PR allows `get_native_function_declarations` API to take a function as argument. This function should take `NativeFunction` as input and emit code for native function declaration. By default it is `dest.compute_native_function_declaration`.

Differential Revision: [D41501838](https://our.internmc.facebook.com/intern/diff/D41501838/)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41501838).
